pkg/registry/client.go

```diff
@@ -209,74 +182,143 @@ func ClientOptPlainHTTP() ClientOption {
+
+	if err := reg.Ping(ctx); err != nil {
+		return fmt.Errorf("authenticating to %q: %w", host, err)
 	}
-	if err := c.authorizer.LoginWithOpts(authorizerLoginOpts...); err != nil {
+
+	if err := c.credentialsStore.Put(ctx, host, cred); err != nil {
 		return err
 	}
+
 	fmt.Fprintln(c.out, "Login Succeeded")
 	return nil
 }
 
 // LoginOptBasicAuth returns a function that sets the username/password settings on login
 func LoginOptBasicAuth(username string, password string) LoginOption {
-	return func(operation *loginOperation) {
-		operation.username = username
-		operation.password = password
+	return func(host string, client *Client) error {
+		client.authorizer.Credential = auth.StaticCredential(host, auth.Credential{Username: username, Password: password})
+		return nil
 	}
 }
 
+// LoginOptBasicAuth returns a function that allows plaintext (HTTP) login
+func LoginOptPlainText() LoginOption {
+	return func(host string, client *Client) error {
+		client.plainHTTP = true
+		return nil
+	}
+}
+
+func ensureTLSConfig(client *auth.Client) (*tls.Config, error) {
+	var transport *http.Transport
+
+	switch t := client.Client.Transport.(type) {
+	case *http.Transport:
+		transport = t
+	case *retry.Transport:
+		switch t := t.Base.(type) {
+		case *http.Transport:
+			transport = t
+		}
+	}
+
+	if transport == nil {
+		// we don't know how to access the http.Transport, most likely the
+		// auth.Client.Client was provided by API user
+		return nil, fmt.Errorf("unable to access TLS client configuration, the provided HTTP Transport is not supported, given: %T", client.Client.Transport)
+	}
+
+	if transport.TLSClientConfig == nil {
+		transport.TLSClientConfig = &tls.Config{}
+	}
+
+	return transport.TLSClientConfig, nil
+}
+
 // LoginOptInsecure returns a function that sets the insecure setting on login
 func LoginOptInsecure(insecure bool) LoginOption {
-	return func(operation *loginOperation) {
-		operation.insecure = insecure
+	return func(_ string, client *Client) error {
+		tlsConfig, err := ensureTLSConfig(client.authorizer)
+
+		if err != nil {
+			return err
+		}
+
+		tlsConfig.InsecureSkipVerify = true
+		return nil
 	}
 }
 
 // LoginOptTLSClientConfig returns a function that sets the TLS settings on login.
 func LoginOptTLSClientConfig(certFile, keyFile, caFile string) LoginOption {
-	return func(operation *loginOperation) {
-		operation.certFile = certFile
-		operation.keyFile = keyFile
-		operation.caFile = caFile
+	return func(_ string, client *Client) error {
+		if (certFile == "" || keyFile == "") && caFile == "" {
+			return nil
+		}
+		tlsConfig, err := ensureTLSConfig(client.authorizer)
+		if err != nil {
+			return err
+		}
+
+		if certFile != "" && keyFile != "" {
+			cert, err := os.ReadFile(certFile)
+			if err != nil {
+				return err
+			}
+
+			key, err := os.ReadFile(keyFile)
+			if err != nil {
+				return err
+			}
+
+			authCert, err := tls.X509KeyPair(cert, key)
```

Comment:
Seems that these can now be replaced by [`LoadX509KeyPair`](https://pkg.go.dev/crypto/tls#LoadX509KeyPair)